frontend/packages/core/src/Input/select.tsx

```diff
@@ -204,41 +213,63 @@ const Select = ({
   const defaultIdx = defaultOption < options.length && defaultOption > 0 ? defaultOption : 0;
   const [selectedIdx, setSelectedIdx] = React.useState(defaultIdx);
 
+  // Flattens all options and sub grouped options for easier retrieval
+  const flatOptions: SelectOption[] = options
+    .map(option => (option.group ? option.group.map(groupOption => groupOption) : option))
+    .reduce((pre: any, cur: any) => pre.concat(cur), []);
+
   React.useEffect(() => {
-    if (options.length !== 0) {
-      onChange && onChange(options[selectedIdx]?.value || options[selectedIdx].label);
+    if (flatOptions.length !== 0) {
+      onChange && onChange(flatOptions[selectedIdx]?.value || flatOptions[selectedIdx].label);
     }
   }, []);
 
   const updateSelectedOption = (event: React.ChangeEvent<{ name?: string; value: string }>) => {
     const { value } = event.target;
-    const optionValues = options.map(o => o?.value || o.label);
-    setSelectedIdx(optionValues.indexOf(value));
+    // handle if selecting a header option
+    if (!value) {
+      return;
+    }
+    setSelectedIdx(flatOptions.findIndex(opt => opt?.value === value || opt?.label === value));
     onChange && onChange(value);
   };
 
   if (options.length === 0) {
```

Comment:
should `flatOptions` be used here as well?

Reply:
Good call out, I've updated it. It technically wouldn't matter but its good to be consistent!